File: abysmal/Galdorin_Visigothe.pl
my $count = 0;

sub EVENT_SPAWN {
	quest::settimer("song",5);
}

sub EVENT_SIGNAL {
	quest::say("Ack, I need somethin' light to clear the taste of salt from my mouth.");
}

sub EVENT_TIMER {
	$count++;
	if ($count == 1) {
		quest::say("On one sunny afternoon, I sawr a dwarf a'wanderin'");
	}
	if ($count == 2) {
		quest::say("He was staggerin' and mumblin' and fallin'.");
	}
	if ($count == 3) {
		quest::say("I caught up and asked him if he needed help,");
	}
	if ($count == 4) {
		quest::say("And he turned and yelled as loud as he's stout:");
	}
	if ($count == 5) {
		quest::say("Git yer flowery songs from my ear,");
	}
	if ($count == 6) {
		quest::say("The light notes will soil me beer!");
	}
	if ($count == 7) {
		quest::say("And if ye don't turn and run fast,");
	}
	if ($count == 8) {
		quest::say("I'll drown yer feet in what I ate last!");
		quest::settimer("song",300);
	}
	if ($count == 9) {
		quest::say("Norrath is so great and so dear");
		quest::settimer("song",5);
	}
	if ($count == 10) {
		quest::say("With all the folk that live here:");
	}
	if ($count == 11) {
		quest::say("Dwarves make lots of ale");
	}
	if ($count == 12) {
		quest::say("High elves are too pale");
	}
	if ($count == 13) {
		quest::say("Trolls are slimy and green");
	}
	if ($count == 14) {
		quest::say("Ogres are very unclean");
	}
	if ($count == 15) {
		quest::say("Gnomes can make things tick");
	}
	if ($count == 16) {
		quest::say("Halflings, masters of the trick");
	}
	if ($count == 17) {
		quest::say("Iksar get by with scales");
	}
	if ($count == 18) {
		quest::say("Vah Shir oft' use their tails");
	}
	if ($count == 19) {
		quest::say("Humans are very plain");
	}
	if ($count == 20) {
		quest::say("Dark elves deal out pain");
	}
	if ($count == 21) {
		quest::say("Frogloks are smarter now");
	}
	if ($count == 22) {
		quest::say("Erudites, the highest brow");
	}
	if ($count == 23) {
		quest::say("Half elves have no city");
	}
	if ($count == 24) {
		quest::say("Wood elves are so pretty");
	}
	if ($count == 25) {
		quest::say("The last Norrathians I'm told");
	} 
	if ($count == 26) {
		quest::say("Are barbarians who love the cold.");
	} 
	if ($count == 27) {
		quest::say("All these folk share these lands,");
	} 
	if ($count == 28) {
		quest::say("Putting it in very good hands.");
		quest::settimer("song",300);
	}
	if ($count == 29) {
		quest::say("In the dark of the fearless night,");
		quest::settimer("song",5);
	}
	if ($count == 30) {
		quest::say("Fantastic stories retold by torchlight.");
	}
	if ($count == 31) {
		quest::say("A tale of a rogue of great prowess.");
	}
	if ($count == 32) {
		quest::say("An enchanter as his revered mistress.");
	}
	if ($count == 33) {
		quest::say("Then there's a shadow knight who found the two,");
	}
	if ($count == 34) {
		quest::say("He swore to them he would always be true.");
	}
	if ($count == 35) {
		quest::say("Some say they were brought together by chance,");
	}
	if ($count == 36) {
		quest::say("Others think that it was circumstance.");
	}
	if ($count == 37) {
		quest::say("No matter how they came to be the closest three,");
	}			
	if ($count == 38) {
		quest::say("They are the greatest adventurers that will ever be.");
                quest::settimer("song",300);
        }
        if ($count == 39){
		quest::say("Tra-la-la and a-diddly-dee.");
                quest::settimer("song",5);
        }
        if ($count == 40) {
                quest::say("The gnome bit the ogre on the knee.");
        }
        if ($count == 41) {
                quest::say("Tra-la-la and a-diddly-dum.");
        }
        if ($count == 42) {
                quest::say("The ogre bit off the gnome's thumb.");
        }
        if ($count == 43) {
                quest::say("Then the ogre said as he turned to me,");
        }
        if ($count == 44) {
                quest::say("Dat fer singin' 'Tra-la-la and a-diddly-dee.'");
                quest::settimer("song",295);
        }
	if ($count == 45) {
		$count = 0;
		quest::settimer("song",5);
	}
}

File: abysmal/Galdorin_Visigothe.lua
-- TODO: Verify

local ThreadManager = require("thread_manager");
local evt;

local songs = {
	[1] = {
		"Tra-la-la and a-diddly-dee.",
		"The gnome bit the ogre on the knee.",
		"Tra-la-la and a-diddly-dum.",
		"The ogre bit off the gnome's thumb.",
		"Then the ogre said as he turned to me,",
		"Dat fer singin' 'Tra-la-la and a-diddly-dee.'"
	},
	[2] = {
		"In the dark of the fearless night,",
		"Fantastic stories retold by torchlight.",
		"A tale of a rogue of great prowess.",
		"An enchanter as his revered mistress.",
		"Then there's a shadow knight who found the two,",
		"He swore to them he would always be true.",
		"Some say they were brought together by chance,",
		"Others think that it was circumstance.",
		"No matter how they came to be the closest three,",
		"They are the greatest adventurers that will ever be."
	},
	[3] = {
		"On one sunny afternoon, I sawr a dwarf a'wanderin'",
		"He was staggerin' and mumblin' and fallin'.",
		"I caught up and asked him if he needed help,",
		"And he turned and yelled as loud as he's stout:",
		"Git yer flowery songs from my ear,",
		"The light notes will soil me beer!",
		"And if ye don't turn and run fast,",
		"I'll drown yer feet in what I ate last!"
	},
	[4] = {
		"Norrath is so great and so dear",
		"With all the folk that live here:",
		"Dwarves make lots of ale",
		"High elves are too pale",
		"Trolls are slimy and green",
		"Ogres are very unclean",
		"Gnomes can make things tick",
		"Halflings, masters of the trick",
		"Iksar get by with scales",
		"Vah Shir oft' use their tails",
		"Humans are very plain",
		"Dark elves deal out pain",
		"Frogloks are smarter now",
		"Erudites, the highest brow",
		"Half elves have no city"
	},
	[5] = {
		"Wood elves are so pretty",
		"The last Norrathians I'm told",
		"Are barbarians who love the cold.",
		"All these folk share these lands,",
		"Putting it in very good hands."
	}
}

function event_spawn(e)
	eq.set_timer("heartbeat", 500); -- 0.5s heartbeat
	ThreadManager:Clear();
	ThreadManager:Create("Sing", sing);
end

function event_timer(e)
	evt = e;
	ThreadManager:Resume("Sing");
end

function event_signal(e)
	if e.signal == 1 then
		e.self:Say("Ack, I need somethin' light to clear the taste of salt from my mouth.");
		eq.signal(279042,12,10 * 1000); -- Galadra Pershan
	elseif e.signal == 2 then
		e.self:Emote("breaks into a wide grin 'Hello, lass!  Why, I've never seen a lovelier woman in all my days!  I'd be grateful if you could bring me your best dark beer.  Hurry back!'");
		eq.signal(279042,12,10 * 1000); -- Galadra Pershan
	end
end

function sing()
	while true do
		local song_number = math.random(1,5); -- Pick random song
		for verse = 1,table.getn(songs[song_number]) do -- iterate through verses
			evt.self:SpellFinished(4724, evt.self);
			evt.self:Say(songs[song_number][verse]);  -- sing verse
			ThreadManager:Wait(2.5);
		end
		ThreadManager:Wait(5 * 60); -- 5 Minutes
	end
end

function event_trade(e)
	local item_lib = require("items");
	if item_lib.check_turn_in(e.trade, {item1 = 9590}) then -- Item: Notice to Cease and Desist
		e.self:Say("Hello. Please pardon me, but. . . Hmm, what do we have here? It says that you wish me to cease and desist spreading slander about the swashbuckler known as Stanos, or else . . .? I'd hardly place him in the 'swashbuckling' category, friend. Let's not sugercoat it. He's more of a scoundrel or smuggler or sniveling scumsucking slimeball. I can't believe you would have the nerve to come in here and threaten my right to say what I wish. My very livelihood depends upon my being able to speak freely to entertain the visitors of this tavern. I'll not be threatened by you, brute. Fight me if you dare!'");
		make_attackable(e.self, true);
		e.self:AddToHateList(e.other, 1);
	end

	item_lib.return_items(e.self, e.other, e.trade)
end

function make_attackable(mob, attackable)
    mob:SetSpecialAbility(SpecialAbility.immune_melee, attackable and 0 or 1)
    mob:SetSpecialAbility(SpecialAbility.immune_magic, attackable and 0 or 1)
    mob:SetSpecialAbility(SpecialAbility.immune_aggro, attackable and 0 or 1)
    mob:SetSpecialAbility(SpecialAbility.immune_aggro_on, attackable and 0 or 1)
    mob:SetSpecialAbility(SpecialAbility.no_harm_from_client, attackable and 0 or 1)
end